R/fwrite.R

```diff
@@ -11,8 +11,12 @@ fwrite = function(x, file="", append=FALSE, quote="auto",
            compress = c("auto", "none", "gzip"),
            yaml = FALSE,
            bom = FALSE,
-           verbose=getOption("datatable.verbose", FALSE)) {
+           verbose=getOption("datatable.verbose", FALSE),
+           encoding = "") {
```

Comment:
nit: could have signature `encoding = c("", "UTF-8", "native")` and use `match.args` instead of the check below

Reply:
The encoding arg of `fread()` doesn't use this signature, either. The reason is the empty string seems not work well with `match.arg()` :

``` r
f <- function(x = c("", "a", "b")) { 
  match.arg(x)
}
f()
#> [1] ""
f("a")
#> [1] "a"
f("b")
#> [1] "b"
f("") # this throws
#> Error in match.arg(x): 'arg' should be one of "", "a", "b"
```

<sup>Created on 2020-10-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Filed the bug to https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17959